src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunCompilationModuleGroupBase.cs

```diff
@@ -0,0 +1,293 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+using Internal.TypeSystem.Interop;
+using ILCompiler.DependencyAnalysis.ReadyToRun;
+using Debug = System.Diagnostics.Debug;
+using System;
+
+namespace ILCompiler
+{
+    public abstract class ReadyToRunCompilationModuleGroupBase : CompilationModuleGroup
+    {
+        protected HashSet<ModuleDesc> _compilationModuleSet;
+        private HashSet<ModuleDesc> _versionBubbleModuleSet;
+        private Dictionary<TypeDesc, ModuleToken> _typeRefsInCompilationModuleSet;
+        private bool _compileGenericDependenciesFromVersionBubbleModuleSet;
+        private ConcurrentDictionary<TypeDesc, bool> _containsTypeLayoutCache = new ConcurrentDictionary<TypeDesc, bool>();
```

Comment:
Would it make sense to move some of these dictionaries to the base class? They don't seem to be specific to a particular compilation mode and may provide a bit of speedup everywhere unless I'm mistaken.